srml/balances/src/lib.rs

```diff
@@ -110,29 +107,34 @@
 //! # fn main() {}
 //!```
 //!
-//! [(gas.rs)](https://github.com/paritytech/substrate/blob/master/srml/contract/src/gas.rs):
+//! The Staking module uses the `LockableCurrency` trait to lock a stash account's funds:
 //!
-//! ```ignore
-//! use srml_support::traits::Currency;
-//! # pub trait Trait: system::Trait {}
+//! ```
+//! use srml_support::traits::{WithdrawReasons, LockableCurrency};
+//! use system;
+//! use primitives::traits::Bounded;
+//! pub trait Trait: system::Trait {
+//! 	type Currency: LockableCurrency<Self::AccountId, Moment=Self::BlockNumber>;
+//! }
+//! # struct StakingLedger<T: Trait> {
+//! # 	stash: <T as system::Trait>::AccountId,
+//! # 	total: <<T as Trait>::Currency as srml_support::traits::Currency<<T as system::Trait>::AccountId>>::Balance,
+//! # 	phantom: std::marker::PhantomData<T>,
+//! # }
+//! # const STAKING_ID: [u8; 8] = *b"staking ";
 //!
-//! pub fn refund_unused_gas<T: Trait>(
-//! 	transactor: &T::AccountId,
-//! 	gas_meter: GasMeter<T>,
-//! 	imbalance: NegativeImbalanceOf<T>,
+//! fn update_ledger<T: Trait>(
+//! 	controller: &T::AccountId,
+//! 	ledger: &StakingLedger<T>
 //! ) {
-//! 	let gas_spent = gas_meter.spent();
-//! 	let gas_left = gas_meter.gas_left();
-//!
-//! 	// Increase total spent gas.
-//! 	<GasSpent<T>>::mutate(|block_gas_spent| *block_gas_spent += gas_spent);
-//!
-//! 	// Refund gas left by the price it was bought at.
-//! 	let refund = <T::Gas as As<BalanceOf<T>>>::as_(gas_left) * gas_meter.gas_price;
-//! 	let refund_imbalance = T::Currency::deposit_creating(transactor, refund);
-//! 	if let Ok(imbalance) = imbalance.offset(refund_imbalance) {
-//! 		T::GasPayment::on_unbalanced(imbalance);
-//! 	}
+//! 	T::Currency::set_lock(
+//! 		STAKING_ID,
+//! 		&ledger.stash,
+//! 		ledger.total,
+//! 		T::BlockNumber::max_value(),
+//! 		WithdrawReasons::all()
+//! 	);
+//! 	// <Ledger<T>>::insert(controller, ledger); // Commented out as we don't have acces to Staking's storage here.
```

Comment:
Maybe use a different example here? Or just drop the last line?

Reply:
Yeah, I looked a lot. The problem is the nature of `Currency` in general, in that if you are locking or transferring funds, you are probably doing a lot of checks and/or writing to storage because the currency event was in response to some other event (like seconding a referendum). I'm OK with removing it.